SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimAnalyzer.cc

```diff
@@ -261,7 +460,522 @@ EcalEBTrigPrimAnalyzer::analyze(const edm::Event& iEvent, const  edm::EventSetup
+
+      */
+
+      treeCl_->Fill();    
+
+    }
+    std::cout << " nCluTPPerEvt " << nCluTPPerEvt << std::endl;
+    hCluTPperEvt_ ->Fill(float(nCluTPPerEvt));
+
+  }
+
   
+  if ( nClu>0)         h_nClu_[0] ->Fill(float(nClu));
+  if ( nClu_Etgt_05>0) h_nClu_[1] ->Fill(float(nClu_Etgt_05));
+  if ( nClu_Etgt_1>0)  h_nClu_[2] ->Fill(float(nClu_Etgt_1));
+  if ( nClu_Etgt_2>0)  h_nClu_[3] ->Fill(float(nClu_Etgt_2));
+  if ( nClu_Etgt_3>0)  h_nClu_[4] ->Fill(float(nClu_Etgt_3));
+  
+  
+  // match the clusters made of RH with gen
+  float nGenInEB=0;    
+  if ( isGenParticleValid_ )  {
+    reco::GenParticleCollection::const_iterator gpIter ;    
+    reco::GenParticleCollection::const_iterator gpIterMatch ;    
+    reco::Candidate::PolarLorentzVector* cluRH_p4;
+    reco::Candidate::PolarLorentzVector* trueEle_p4=0;
+    reco::Candidate::PolarLorentzVector* matchEle_p4=0;
+    BaseParticlePropagator prop;
+    
+    float dR=9999;
+    nGenInEB=0;    
+    
+    
+    for ( std::vector<PhotonMCTruth>::const_iterator imcEle=mcEle.begin(); imcEle!=mcEle.end(); imcEle++ ) {
+      
+      if (   abs(imcEle->fourMomentum().pseudoRapidity()) >= 1.5 || imcEle->fourMomentum().et() < 10 ) continue;
+      //      if ( imcEle->hasBrem() !=0) continue;
+      
+      if (debug_) std::cout << " imcEle pt " << imcEle->fourMomentum().et() << " E " << imcEle->fourMomentum().e() << std::endl;
+      bool isTheSame=false;
+      for ( gpIter=gp->begin() ; gpIter!=gp->end() ; gpIter++ ) {
+	//	if ( abs(gpIter->pdgId()) != 11 ) continue;
+	// float ch= gpIter->pdgId()/11;
+	if ( abs(gpIter->pdgId()) != 22 ) continue;
+	float ch= 0.;
+
+        
+
+	if (debug_) std::cout << "gpIter charge in the loop " << ch <<  " and Pt " << gpIter->pt() << std::endl;
+
+	isTheSame=false;
+	
+	float myMCPhi=imcEle->fourMomentum().phi();
+	myMCPhi= normalizedPhi(myMCPhi);
+
+	float gpPhi=gpIter->phi();
+	gpPhi= normalizedPhi(gpPhi);
+	double dPhi = myMCPhi - gpPhi;
+	std::cout << " myMCPhi " << myMCPhi << " gpPhi " << gpPhi << std::endl;
+	double dEta = fabs(imcEle->fourMomentum().pseudoRapidity() - gpIter->eta());
+	double dPt = fabs(imcEle->fourMomentum().et() - gpIter->pt() );
+	
+	if ( dEta <= 0.0001 && dPhi <= 0.0001 && dPt <= 0.0001)  {
+	  isTheSame=true;
+          gpIterMatch=gpIter;
+	  break;
+	}
+
+      }
+
+      if ( ! isTheSame ) continue;
+      
+      //  std::cout << " mctruth eloss  " << imcEle->eloss()[0] << " fBrem " << imcEle->eloss()[0]/imcEle->fourMomentum().e() << std::endl;     
+      ///h_fBrem_truth_->Fill( imcEle->eloss()[0]/imcEle->fourMomentum().e() );
+      nGenInEB++;	    
+      reco::Candidate::LorentzVector eleP4(imcEle->fourMomentum().px(),imcEle->fourMomentum().py(), imcEle->fourMomentum().pz(), imcEle->fourMomentum().e()); 
+      RawParticle trueEle(eleP4);
+      trueEle.setVertex(imcEle->primaryVertex().x(), imcEle->primaryVertex().y(), imcEle->primaryVertex().z(), 0.);
+      //float ch= gpIterMatch->pdgId()/11;
```

Comment:
Delete commented out code, or justify with a comment why you still need to keep it